src/app/_components/common/datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,73 @@
+"use client";
+
+import type { DateTimePickerProps } from "@mantine/dates";
+import { DateTimePicker } from "@mantine/dates";
+import styles from "./DatePicker.module.scss";
+
+export interface DatePickerProps extends Omit<DateTimePickerProps, "value" | "onChange"> {
+  label: string;
+  placeholder?: string;
+  required?: boolean;
+  error?: string;
+  value?: string | null;
+  onChange?: (value: string | null) => void;
+}
+
+/**
+ * DatePicker component
+ *
+ * Pick a date and time, returns an iso string value
+ *
+ * @example
+ * const [date, setDate] = useState<string | null>(null);
+ *
+ * <DatePicker
+ *   label="Departure Time"
+ *   value={date}
+ *   onChange={setDate}
+ * />
+ *
+ */
+export default function DatePicker({
+  label,
+  placeholder,
+  required = false,
+  error,
+  valueFormat = "DD MMM YYYY hh:mm A",
+  value,
+  onChange,
+  ...props
+}: DatePickerProps) {
+  // Convert ISO string to Date object for Mantine
+  const mantineValue = value ? new Date(value) : null;
```

Comment:
maybe we can have some error handling for bad iso strings

```suggestion
     const mantineValue = value ? new Date(value) : null;
   // Could add validation:
   if (value && (mantineValue === null || isNaN(mantineValue.getTime()))) {
     console.warn('Invalid ISO string provided to DatePicker:', value);
     return null;
   }
```